.github/workflows/style-checks.yml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2021-2024 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+name: Coding style validation 
+
+on: [pull_request]
+
+env:
+  PR_NUMBER: ${{ github.event.number }}
+  GH_TOKEN: ${{ github.token }}
+
+permissions:
+  contents: read
+
+jobs:
+  check-style:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        path: openssl
+    - name: check style for each commit 
+      working-directory: openssl
+      shell: bash
+      run: |
+        ERRORS_FOUND=0
+        git fetch origin $GITHUB_BASE_REF:$GITHUB_BASE_REF
+        REFSTART=$(git rev-parse $GITHUB_BASE_REF)
+        REFEND=$(git rev-parse HEAD)
+        echo "Checking from $REFSTART to $REFEND"
+        for i in $(git log --no-merges --format=%H $REFSTART..$REFEND)
+        do
+          echo "Checking commit $i"
+          set +e
+          ./util/check-format-commit.sh $i
+          if [ $? -ne 0 ]
+          then
+            ERRORS_FOUND=1
+          fi
+          set -e
+        done
+        SKIP_TEST=$(gh pr view $PR_NUMBER --json labels --jq '.labels[] | select(.name == "style:exempted") | .name')
```

Comment:
I'd prefer `style: waived` or something like that.

Reply:
I'm not opposed, but can I ask why?  I used style:exempted because we already have tests:exempted, and I was trying to stick with the pattern

Reply:
My reasoning was that you would apply the label _after_ the test runs and fails, you review the results and see that the failures detected are either false positives or that it would be inappropriate to fix them in the PR for other reasons. So you basically waive the failure.

But I do not insist on this name, however at least add a space after `:` so it has the same format as the other labels with `:` in them.

Reply:
I'll change it, the  reasoning makes sense, just didn't previously understand the justification